dot/peerset/peerset.go

```diff
@@ -530,24 +542,26 @@ func (ps *PeerSet) removePeer(setID int, peers ...peer.ID) error {
 		}
 
 		if status := ps.peerState.peerStatus(setID, pid); status == connectedPeer {
-			ps.resultMsgCh <- Message{
+			dropMessage := Message{
 				Status: Drop,
 				setID:  uint64(setID),
 				PeerID: pid,
 			}
 
+			ps.processor.Process(dropMessage)
+
 			// disconnect and forget
 			err := ps.peerState.disconnect(setID, pid)
 			if err != nil {
-				return err
+				return fmt.Errorf("cannot disconnect peer %s at set %d: %w", pid, setID, err)
```

Comment:
nit maybe `from` instead of `at? Maybe not though

```suggestion
				return fmt.Errorf("cannot disconnect peer %s from set %d: %w", pid, setID, err)
```